event-driven-generated/src/event.rs

```diff
@@ -0,0 +1,90 @@
+use ratatui::crossterm::event::{self, Event as CrosstermEvent};
+use std::sync::mpsc;
+use std::thread;
+use std::time::{Duration, Instant};
+
+/// Application events.
+///
+/// You can extend this enum with your own custom events.
+#[derive(Clone, Debug)]
+pub enum AppEvent {
+    /// Custom increment event.
+    Increment,
+    /// Custom decrement event.
+    Decrement,
+    /// Quit the application.
+    Quit,
+}
+
+/// Representation of all possible events.
+#[derive(Clone, Debug)]
+pub enum Event {
+    /// An event that is emitted on a regular schedule.
+    ///
+    /// Use this event to run any code which has to run outside of being a direct response to
+    /// a user event. e.g. polling exernal systems, updating animations, or rendering the UI
+    /// based on a fixed frame rate.
```

Comment:
It could be worth also noting / mentioning the same in the tick method. Also make a note of the FPS that the template uses (looks like 30 below)